assets/src/editor/blocks/share-article/index.js

```diff
@@ -37,36 +42,122 @@ export const settings = {
 			type: 'boolean',
 			default: true,
 		},
+		enableLinkedIn: {
+			type: 'boolean',
+			default: true,
+		},
+		enableEmail: {
+			type: 'boolean',
+			default: true,
+		},
+		enableCopyLink: {
+			type: 'boolean',
+			default: true,
+		},
 	},
 
 	example: {},
 
 	/**
-	 * Edit component used to manage featured image and page intro.
+	 * Edit component for managing social share settings.
 	 */
 	edit: function ShareArticleBlock( { attributes, setAttributes } ) {
 		const {
 			enableTwitter,
 			enableFacebook,
+			enableLinkedIn,
+			enableEmail,
+			enableCopyLink,
 		} = attributes;
 
 		const blockProps = useBlockProps( {
 			className: 'share-article',
 		} );
 
 		return (
-			<div { ...blockProps } >
-				{ ( ! enableTwitter && ! enableFacebook ) && (
+			<div { ...blockProps }>
+				{ ( ! enableTwitter && ! enableFacebook && ! enableLinkedIn && ! enableEmail && ! enableCopyLink ) && (
```

Comment:
Doesn't need to be changed, this logic is 💯 , but had a hypothetical think about how we'd make this less list-all-the-services agnostic

```js
const enabledServiceCount = Object.keys( attributes ).reduce(
    ( enabledCount, attrName ) => {
        if ( ! /^enabled/.test( attrName ) ) {
            return enabledCount;
        }
        return enabledCount + ( attributes[ attrName ] ? 1 : 0 );
    },
    0
);
```
then we could make this type of conditional `{ enabledServiceCount === 0 ....`
Advantage: assuming all keys are well-named and consistently set, lets you summarize things in a service-independent way. Disadvantage: bakes in assumptions about key names for other unrelated attributes.
**No change needed**, just Friday daydreams